TESTS/mbed_drivers/flashiap/main.cpp

```diff
@@ -81,6 +81,53 @@ void flashiap_program_test()
     TEST_ASSERT_EQUAL_INT32(0, ret);
 }
 
+void flashiap_cross_sector_program_test()
+{
+    FlashIAP flash_device;
+    uint32_t ret = flash_device.init();
+    TEST_ASSERT_EQUAL_INT32(0, ret);
+
+    uint32_t page_size = flash_device.get_page_size();
+
+    // Erase last two sectors
+    uint32_t address = flash_device.get_flash_start() + flash_device.get_flash_size();
+    uint32_t sector_size;
+    for (uint32_t i = 0; i < 2; i++) {
+        sector_size = flash_device.get_sector_size(address - 1UL);
+        TEST_ASSERT_NOT_EQUAL(0, sector_size);
+        TEST_ASSERT_TRUE(sector_size % page_size == 0);
+
+        address -= sector_size;
+        ret = flash_device.erase(address, sector_size);
+        TEST_ASSERT_EQUAL_INT32(0, ret);
+    }
+
+    address += sector_size - 3 * page_size;
+    uint32_t aligned_prog_size = 7 * page_size;
```

Comment:
some devices might fail to allocate 7*page_size (if we consider that data are deleted on line 124, this means we are allocating double than that, results in 14*page_size?

Lets review the test results for this test

Reply:
Maybe I'm mistaken, but AFAIK page sizes are typically 8 bytes. Is this problematic?

Reply:
I recall there was a problem but was probably with sectors rather than pages (as page size that is returned is the smallest unit thta can be written, thus in most cases would be 8 bytes).

Just checking that this will be ok for all targets

Reply:
Appears you were right. Boards (namely ARCH_PRO) have a 1KB page size. Modifying the test accordingly.